crates/turborepo-lib/src/client.rs

```diff
@@ -0,0 +1,127 @@
+use std::{env, future::Future};
+
+use anyhow::{anyhow, Result};
+use axum::async_trait;
+use lazy_static::lazy_static;
+use reqwest::StatusCode;
+use serde::Deserialize;
+
+use crate::{get_version, retry::retry_future};
+
+#[async_trait]
+pub trait UserClient {
+    fn set_token(&mut self, token: String);
+    async fn get_user(&self) -> Result<UserResponse>;
+}
```

Comment:
I'm assuming this trait is here for testing, but I'm not seeing any use this. Will we need to refactor the `login` to use this trait before we can add tests?